src/main/java/com/google/devtools/build/lib/exec/SpawnInputExpander.java

```diff
@@ -119,7 +122,17 @@ void addRunfilesToInputs(
           if (strict && !actionFileCache.getMetadata(localArtifact).getType().isFile()) {
             throw new IOException("Not a file: " + localArtifact.getPath().getPathString());
           }
-          addMapping(inputMap, location, localArtifact);
+
+          if (localArtifact.isTreeArtifact()) {
```

Comment:
Does this actually need a special case? It seems like the old code would simply create a symlink to a tree artifact directory, which should work just as well as symlinking individual files, right?

Reply:
You're right that that would work for local execution, but it wouldn't work for remote execution.